File: drone-plugin/lib-ts/summary/summary-upload.model.ts

import { Repository } from "../repo/repo.model"
import { ArtifactUpload } from "../artifact/artifact-upload.model"

interface SummaryUpload {
  Repository: Repository // Assign this yourself
  BuildID: number // 2
  BranchID: string // "feature/no-more-bugs"
  Artifacts: ArtifactUpload[] // Files you want to attach
  Commit: string // Commit hash
  Success: boolean
  Created: Date // Date Summary was recorded
  Message?: string
  Author?: string
}

export { SummaryUpload }

File: server/src/upload/summary-upload.model.go
package upload

import (
	"github.com/ZombieHippie/test-gin/server/src/repo"
	"time"
)

// SummaryUpload is a common object for reports to point at
type SummaryUpload struct {
	Repository repo.Repository
	BranchID   string
	BuildID    int
	Artifacts  []ArtifactUpload
	Commit     string

	Message string
	Author  string
	Success bool
	Created time.Time
}
